### ISSUE TITLE
Awaiting stops working in Editor after loading scene

### ISSUE THREAD

**Issue by cfoulston** [muckSponge/UnityAsync#13]
I believe this is because of the commit here:
https://github.com/muckSponge/UnityAsync/commit/e01e2dc39084e4b5181dcef98344291ea3d5ddc3#diff-86c1eb8e1ae625825c97d335eaea9712

`if(!Application.isEditor) // DontDestroyOnLoad can not be called in editor mode
				DontDestroyOnLoad(Instance);`

When loading a new scene, the AsyncManager instance is destroyed preventing any Await calls to complete.

DontDestroyOnLoad does work in Editor. So not following the reasoning behind the change.

**Comment by drallcom3** [comment]
Same here. UnityAsync fails heavily after loading a scene. Removing the IF makes it work fine.
Mainly the frame counter becomes unusable.

**Comment by aylanonsense** [comment]
Same! Resolved by commenting out the conditional

**Comment by tim-pp** [comment]
+1, I wonder what the reason is behind that line.

**Comment by prodigga** [comment]
Pinging @muckSponge, just because some information would be useful. I am willing to put in a pull request to fix this but it would be good to understand why this code and the comment exists in the first place.

**Comment by muckSponge** [comment]
Yeah go ahead with the pull request. If DontDestroyOnLoad works in the editor, that condition can be removed. I probably just didn't do thorough enough testing when I approved it.

**Comment by muckSponge** [comment]
Actually, decided to fix this myself in cfd37ae62254fe35c893a9f73dcffe1284c453e2.
